h2o-algos/src/main/java/hex/tree/TreeHandler.java

```diff
@@ -182,7 +182,7 @@ private static void fillnodeDescriptions(final SharedTreeNode node, final String
         int[] nodeLevels = node.getParent().isBitset() ? extractNodeLevels(node) : null;
         nodeDescriptionBuilder.append("Node has id ");
         nodeDescriptionBuilder.append(node.getNodeNumber());
-        if (node.getColName() != null) {
+        if (node.getColName() != null && node.getRightChild() == null && node.getLeftChild() == null) {
```

Comment:
`SharedTreeNode` has a `isLeaf` method

Reply:
Thanks, will fix ASAP.

Reply:
Fixed in https://github.com/h2oai/h2o-3/pull/4340/commits/7895038b96793e2991f9a8b87e85725f177bba9a